CONTRIBUTING.md

```diff
@@ -1,32 +1,90 @@
+####INDEX
+* [Pull Requests](#pull-requests)
+* [Setting Up Your Environment](#setting-up-your-environment)
+  * [Assumptions](#assumptions)
+  * [Your Repository](#setting-up-your-repository)
+  * [Solution Dependencies](#setting-up-the-solution-dependencies)
+
 Pull Requests
 =============
 
+* Please reference [Style Guide](STYLEGUIDE.md)
 * All PRs must be related to an open issue
```

Comment:
I'd say append a second sentence here: "If you want to make a PR for something that doesn't have an issue yet, do it by first making a new issue for it, then submitting the PR."

Make it clear that we're not trying to say "only work on the already identified stuff".